### PR TITLE
Allow underscores when initlalising BigInt

### PR DESCRIPTION
**Pull request by alex-kampa**
Implements https://github.com/crystal-lang/crystal/issues/7106

### CODE REVIEW

**Comment by Sija** [comment]
Duplicate of #7107

**Comment by bcardiff** [comment]
It’s duplicate but this pr adds some docs :-)

Luckily we don’t need to discuss the implementation.

Thank you both for the contribution.

File: spec/std/big/big_int_spec.cr [hunk 1]
```diff
@@ -24,6 +24,7 @@ describe "BigInt" do
     BigInt.new("12345678").to_s.should eq("12345678")
     BigInt.new("+12345678").to_s.should eq("12345678")
     BigInt.new("-12345678").to_s.should eq("-12345678")
+    BigInt.new("-1234_5678_0000").to_s.should eq("-123456780000")
```
Comment (Sija) [inline spec/std/big/big_int_spec.cr]:
```suggestion
    BigInt.new("-123_456_780_000").to_s.should eq("-123456780000")
```

**Comment by bcardiff** [comment]
Closing since #7107 ended up implementing the same feature for other Big numbers (decimal, floats).

**Pull request closed without merging**